src/torch_onnx/_patch.py

```diff
@@ -50,39 +50,46 @@ def _from_dynamic_axes_to_dynamic_shapes(
     if dynamic_axes is None:
         return None
 
-    input_names_set = set() if input_names is None else set(input_names)
-
-    dynamic_shapes = {}
+    if input_names is not None:
+        sig = _signature(model)
+        if len(input_names) > len(sig.parameters):
+            raise ValueError(
+                f"Number of input names ({len(input_names)}) should not be greater than the number of model inputs ({len(sig.parameters)})"
+            )
+        input_names_to_model_inputs = {
+            input_name: param_name
+            for input_name, param_name in zip(input_names, sig.parameters)
+        }
+
+    # NOTE: torch.export.export does not support input names assignment,
+    # so we need to map input names to model inputs to create dynamic_shapes
+    # for the exported program
+    dynamic_shapes_to_exported_program = {}
     for input_name, axes in dynamic_axes.items():
-        if input_name in input_names_set:
+        if input_name not in input_names_to_model_inputs:
```

Comment:
When input_names is None, will this always raise the ValueError below? (also input_names_to_model_inputs looks to be uninitialized when input_names is None)